game.py

```diff
@@ -407,6 +412,32 @@ def on_update(self):
                     self.board.overlay.update_unit(self.players[0].units[new_idx])
                 self.map_pos = (clamp(self.board.selected_unit.location[0] - 12, -1, 77),
                                 clamp(self.board.selected_unit.location[1] - 11, -1, 69))
+        elif pyxel.btnp(pyxel.KEY_M):
+            units = self.players[0].units
+            filtered_units = [unit for unit in units if not unit.besieging and unit.remaining_stamina > 0]
+
+            if self.game_started and self.board.overlay.can_iter_settlements_units() and \
+                    len(filtered_units) > 0:
+                self.board.overlay.remove_warning_if_possible()
+                if self.board.overlay.is_setl():
+                    self.board.selected_settlement = None
+                    self.board.overlay.toggle_settlement(None, self.players[0])
+                if self.board.selected_unit is None or isinstance(self.board.selected_unit, Heathen):
+                    self.board.selected_unit = filtered_units[0]
+                    self.board.overlay.toggle_unit(filtered_units[0])
+                elif len(filtered_units) > 1:
```

Comment:
I believe this will need to be an `else` instead of an `elif`, because if you have two movable units, and you move one, and you are left with only one movable, you should be able to press M to get to the last one. But because your currently selected unit is no longer in `filtered_units`, and its length is 1, nothing happens.

Reply:
Ah true forgot about that. I will fix that up now.

Reply:
Addressed in ac9a57469583e0c937ce2183dccf6c3066573106